CONTRIBUTING.md

```diff
@@ -1,25 +1,67 @@
 # Developer Guide
 
-# TODO (andreyvelich): This doc needs to be updated for Kubeflow Trainer V2
-
-Kubeflow Training Operator is currently at v1.
+This guide explains how to contribute to the Kubeflow Trainer V2 project.
 
 ## Requirements
 
 - [Go](https://golang.org/) (1.23 or later)
-- [Docker](https://docs.docker.com/) (23 or later)
+- [Docker](https://www.docker.com/) (20.10.0 or later)
 - [Python](https://www.python.org/) (3.11 or later)
 - [kustomize](https://kustomize.io/) (4.0.5 or later)
-- [Kind](https://kind.sigs.k8s.io/) (0.22.0 or later)
+- [Kind](https://kind.sigs.k8s.io/) (0.27.0 or later)
 - [Lima](https://github.com/lima-vm/lima?tab=readme-ov-file#adopters) (an alternative to DockerDesktop) (0.21.0 or later)
   - [Colima](https://github.com/abiosoft/colima) (Lima specifically for MacOS) (0.6.8 or later)
```

Comment:
Let's move Lima and Colima right after Docker since it is alternative to Docker Desktop
```suggestion
- [Docker]....
- [Lima](https://github.com/lima-vm/lima?tab=readme-ov-file#adopters) (an alternative to DockerDesktop) (0.21.0 or later)
  - [Colima](https://github.com/abiosoft/colima) (Lima specifically for MacOS) (0.6.8 or later)
```